File: Punto4/punto4.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

struct Estudiante {
    string nombre;
    int edad;
    float promedio;
};

int main() {
    vector<Estudiante> estudiantes = {
        {"Ana Torres", 21, 4.2},
        {"Luis Mendoza", 19, 3.8},
        {"Sofia Castro", 22, 4.9}
    };

    cout << "--- Antes de la modificacion ---" << endl;
    for (const auto& est : estudiantes) {
        cout << est.nombre << ": " << est.promedio << endl;
    }

    // Cambiar el promedio de un estudiante específico (Luis Mendoza en el índice 1)
    cout << "\nModificando el promedio de " << estudiantes[1].nombre << "..." << endl;
    estudiantes[1].promedio = 4.5;

    cout << "\n--- Despues de la modificacion ---" << endl;
    for (const auto& est : estudiantes) {
        cout << est.nombre << ": " << est.promedio << endl;
    }

    return 0;
}

File: Punto4/punto4.py
from dataclasses import dataclass

@dataclass
class Estudiante:
    nombre: str
    edad: int
    promedio: float

def main():
    estudiantes = [
        Estudiante("Ana Torres", 21, 4.2),
        Estudiante("Luis Mendoza", 19, 3.8),
        Estudiante("Sofia Castro", 22, 4.9)
    ]

    print("--- Antes de la modificacion ---")
    for est in estudiantes:
        print(f"{est.nombre}: {est.promedio}")

    # Cambiar el promedio de un estudiante específico (Luis Mendoza)
    # Buscando por nombre para hacerlo más dinámico
    for est in estudiantes:
        if est.nombre == "Luis Mendoza":
            print(f"\nModificando el promedio de {est.nombre}...")
            est.promedio = 4.5
            break

    print("\n--- Despues de la modificacion ---")
    for est in estudiantes:
        print(f"{est.nombre}: {est.promedio}")

if __name__ == "__main__":
    main()
